src/wallet/api/wallet.cpp

```diff
@@ -1306,89 +1253,70 @@ PendingTransaction *WalletImpl::createSweepUnmixableTransaction()
 
         } catch (const tools::error::daemon_busy&) {
             // TODO: make it translatable with "tr"?
-            m_errorString = tr("daemon is busy. Please try again later.");
-            m_status = Status_Error;
+            setStatusError(tr("daemon is busy. Please try again later."));
         } catch (const tools::error::no_connection_to_daemon&) {
-            m_errorString = tr("no connection to daemon. Please make sure daemon is running.");
-            m_status = Status_Error;
+            setStatusError(tr("no connection to daemon. Please make sure daemon is running."));
         } catch (const tools::error::wallet_rpc_error& e) {
-            m_errorString = tr("RPC error: ") +  e.to_string();
-            m_status = Status_Error;
+            setStatusError(tr("RPC error: ") +  e.to_string());
         } catch (const tools::error::get_random_outs_error&) {
-            m_errorString = tr("failed to get random outputs to mix");
-            m_status = Status_Error;
-
+            setStatusError(tr("failed to get random outputs to mix"));
         } catch (const tools::error::not_enough_unlocked_money& e) {
-            m_status = Status_Error;
+            setStatusError("");
```

Comment:
actually no. in previous version of code there was something like this:
```
m_status = Status_Error;
....
m_status = Status_Error;
m_errorString = "...";
```
because api lib wanted a caller to know error happened before setting actual error string. I followed the same way since `m_status` was atomic